src/OrbitClientModel/include/OrbitClientModel/CaptureSerializer.h

```diff
@@ -44,42 +44,45 @@ orbit_client_protos::CaptureInfo GenerateCaptureInfo(
     const absl::flat_hash_map<uint64_t, std::string>& key_to_string_map);
 
 template <class TimersIterator>
-void Save(std::ostream& stream, const CaptureData& capture_data,
+void Save(google::protobuf::io::CodedOutputStream* coded_output, const CaptureData& capture_data,
           const absl::flat_hash_map<uint64_t, std::string>& key_to_string_map,
           TimersIterator timers_iterator_begin, TimersIterator timers_iterator_end) {
-  google::protobuf::io::OstreamOutputStream out_stream(&stream);
-  google::protobuf::io::CodedOutputStream coded_output(&out_stream);
-
   orbit_client_protos::CaptureHeader header;
   header.set_version(kRequiredCaptureVersion);
-  WriteMessage(&header, &coded_output);
+  WriteMessage(&header, coded_output);
 
   orbit_client_protos::CaptureInfo capture_info =
       GenerateCaptureInfo(capture_data, key_to_string_map);
-  WriteMessage(&capture_info, &coded_output);
+  WriteMessage(&capture_info, coded_output);
 
   // Timers
   for (auto it = timers_iterator_begin; it != timers_iterator_end; ++it) {
-    WriteMessage(&(*it), &coded_output);
+    WriteMessage(&(*it), coded_output);
   }
 }
 
 }  // namespace internal
 
 template <class TimersIterator>
-ErrorMessageOr<void> Save(const std::string& filename, const CaptureData& capture_data,
+ErrorMessageOr<void> Save(const std::filesystem::path& filename, const CaptureData& capture_data,
                           const absl::flat_hash_map<uint64_t, std::string>& key_to_string_map,
                           TimersIterator timers_iterator_begin,
                           TimersIterator timers_iterator_end) {
-  std::ofstream file(filename, std::ios::binary);
-  if (file.fail()) {
-    ERROR("Saving capture in \"%s\": %s", filename, "file.fail()");
-    return ErrorMessage("Error opening the file for writing");
+  auto fd_or_error = orbit_base::OpenFileForWriting(filename);
+
+  if (fd_or_error.has_error()) {
+    std::string error_message = absl::StrFormat("Failed to open \"%s\" for writing: %s",
+                                                filename.string(), fd_or_error.error().message());
```

Comment:
That's similar to the situation in the other PR. The error message is kind of redundant:
```
Failed to open "filename.xyz" for writing: Unable to open file "filename.xyz": <strerror msg>

Reply:
done.